fedimint-client/src/lib.rs

```diff
@@ -1991,7 +2091,105 @@ impl Client {
             }
         })
     }
+
+    pub async fn log_event<E>(&self, module_id: Option<ModuleInstanceId>, event: E)
+    where
+        E: Event + Send,
+    {
+        let mut dbtx = self.db.begin_transaction().await;
+        self.log_event_dbtx(&mut dbtx, module_id, event).await;
+        dbtx.commit_tx().await;
+    }
+
+    pub async fn log_event_dbtx<E, Cap>(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_, Cap>,
+        module_id: Option<ModuleInstanceId>,
+        event: E,
+    ) where
+        E: Event + Send,
+        Cap: Send,
+    {
+        dbtx.log_event(self.log_ordering_wakeup_tx.clone(), module_id, event)
+            .await;
+    }
+
+    pub async fn log_event_raw_dbtx<Cap>(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_, Cap>,
+        kind: EventKind,
+        module: Option<(ModuleKind, ModuleInstanceId)>,
+        payload: Vec<u8>,
+    ) where
+        Cap: Send,
+    {
+        let module_id = module.as_ref().map(|m| m.1);
+        let module_kind = module.map(|m| m.0);
+        dbtx.log_event_raw(
+            self.log_ordering_wakeup_tx.clone(),
+            kind,
+            module_kind,
+            module_id,
+            payload,
+        )
+        .await;
+    }
+
+    pub async fn handle_events<F, R, K, V>(&self, pos_key: &K, call_fn: F) -> anyhow::Result<()>
+    where
+        K: DatabaseKey + DatabaseRecord + MaybeSend + MaybeSync,
+        K: DatabaseRecord<Value = V>,
+        V: Into<EventLogId> + From<EventLogId> + Default + MaybeSend + MaybeSync,
+        F: Fn(EventLogEntry) -> R,
```

Comment:
I just copied `autocommit`. We can try to change it if the need ever arises. `FnMut` might have some extra complications w.r.t async, borrowing etc. too, so it might backfire as well.